dwave/optimization/_model.pyi

```diff
@@ -101,6 +101,7 @@ class ArraySymbol(Symbol):
 
     def __iadd__(self, rhs: ArraySymbol) -> NaryAdd: ...
     def __imul__(self, rhs: ArraySymbol) -> NaryMultiply: ...
+    def __iter__(self) -> typing.Iterator[ArraySymbol]: ...
```

Comment:
In practice this will always be `BasicIndexing` for now. But let's leave the door open for the future.